healthcare.management.system/rest-module/src/main/java/com/healthcare/management/system/rest/controller/UserController.java

```diff
@@ -0,0 +1,64 @@
+package com.healthcare.management.system.rest.controller;
+
+import com.healthcare.management.system.rest.entity.User;
+import com.healthcare.management.system.rest.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
+// TODO: 18.08.22 add version in the api endpoint path
+
+public class UserController {
+    private UserService userService;
+
+    @Autowired
+    public  UserController(UserService theUserService) {
+        userService = theUserService;
+    }
+
+    @GetMapping("/users")
+    public List<User> findAllUsers(){
+        return userService.findAll();
+    }
+
+    @GetMapping("/users/{userId}")
+    public User getUser(@PathVariable int userId){
+        User theUser = userService.findById(userId);
+
+        if (theUser == null) {
+            throw new RuntimeException("There is no user with id: " + userId);
+        }
+        return theUser;
+    }
+
+    @PostMapping("/users")
+    public User createUser(@RequestBody User theUser){
+        // just in case they pass an id in JSON ... set id to 0
+        // this is to force a save of new item ... instead of update
+        theUser.setId(0);
```

Comment:
There is another way to handle this situation. Instead of receiving `User` entity class, you can create a `UserDTO ` DTO. Then you need to create a method which maps `UserDTO ` -> `User`.

This will help when to add validation on the user input json data in the web layer, without coupling it inside the `User` entity class which relates to db layer